system/Pager/PagerRenderer.php

```diff
@@ -117,7 +117,7 @@ public function getPrevious()
 
 		$uri = clone $this->uri;
 
-		$uri->addQuery('page', $this->first - 1);
+		$uri->addQuery('page', $this->current - 1);
```

Comment:
Actually, looking at this again, this doesn't work. In the context of this class, `getPrevious` doesn't apply to the page immediately before the current page. It means the link for the "previous" button's link. If you're on page 2 that might be page 1, but if you're on page 10, and you're showing 2 links either side of the current page, that would be page 7.  This matches the way that CodeIgniter has done it in the past. So any changes to `hasPrevious`, `getPrevious`, `hasNext`, and `hasPrevious` will need to be reverted.